package/gluon-statistics-mcast/src/respondd.c

```diff
@@ -0,0 +1,478 @@
+/*
+  Copyright (c) 2021, Linus Lüssing <linus.luessing@c0d3.blue>
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+#include <respondd.h>
+
+#include <json-c/json.h>
+#include <libgluonutil.h>
+
+#include <dirent.h>
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#define BPFCOUNTD_SOCKS_DIR "/var/run/bpfcountd/gluon-sockets"
+
+enum capdir {
+	CAP_IN,
+	CAP_OUT,
+	__CAP_COUNT,
```

Comment:
All identifiers starting with `__` or `_`+uppercase letter are reserved for the compiler and libc and should never be defined in program code.

Reply:
Oh, didn't know about that. Are you sure? Because I see it in so many places for enums, like in netifd (__DEV_ATTR_MAX,  __BOND_ATTR_MAX, __BRIDGE_ATTR_MAX, ...). Or even in the Linux kernel (__NL80211_ATTR_AFTER_LAST, __IFLA_BRIDGE_VLAN_TUNNEL_MAX, ...).

Reply:
In the Linux kernel it's not strictly incorrect, as the kernel is basically its own libc and can therefore use the reserved namespace. For userspace code, it is definitely incorrect according to the C standard (see https://en.cppreference.com/w/c/language/identifier for a summary of the rules).